cyipopt/scipy_interface.py

```diff
@@ -35,6 +35,8 @@
 
 import cyipopt
 
+from scipy.optimize._numdiff import approx_derivative
```

Comment:
I worry a little if this function should be used, as `_numdiff` is not a "public" module.

Reply:
I agree with @moorepants here in principle. Although using finite differencing to get an accurate derivative approximation is complex as an error analysis is required to chose an optimal stepsize that balances truncation and subtractive cancellation. Tapping in to `scipy.optimize` does seem like the best way to do it without otherwise reimplementing ourselves or introducing an additional dependency.

Reply:
There are sevaral problems I encountered. As can be written [here](https://github.com/scipy/scipy/pull/10673), scipy decided to use `approx_derivative` instead of `approx_fprime`. 

This has several disadvantages I want to discuss. As you see from the failed pipeline this is not the best function for approximating a numerical derivative. When testing it on my system I had installed scipy==1.4.1. In this version the change was no incorporated. When upgrading to scipy==1.6.1 the tests are failing too.

Led by these observation I've used the implementation given in [optpy](https://github.com/matthias-k/optpy/blob/master/optpy/jacobian.py). If you are willing to review that I will make a new pull request.

But we have to be aware that the results of the test cases are depending on the actual scipy version when no jacobian function is given!